### PR TITLE
Added integration tests for KFS REST inference and model metadata

### PR DESCRIPTION
**Pull request by bstrzele**

### CODE REVIEW

**Review by atobiszei — APPROVED** [review]

File: src/test/kfs_rest_test.cpp [hunk 1]
```diff
@@ -63,9 +63,9 @@ class MockedServer : public Server {
 
     std::unique_ptr<Module> createModule(const std::string& name) override {
         if (name == ovms::SERVABLE_MANAGER_MODULE_NAME)
-            return std::make_unique<MockedServableManagerModule>();
+            return std::make_unique<ovms::ServableManagerModule>();
         if (name == ovms::GRPC_SERVER_MODULE_NAME)
-            return std::make_unique<MockedGRPCServerModule>(*this);
+            return std::make_unique<ovms::GRPCServerModule>(*this);
```
Comment (atobiszei) [inline src/test/kfs_rest_test.cpp:68]: If we don't use those Mocks then we can delete this class as well and start ovms::Server? Or are there different impediments?

File: src/test/kfs_rest_test.cpp [hunk 2]
```diff
@@ -183,7 +183,70 @@ TEST_F(HttpRestApiHandlerTest, dispatchReady) {
...
+    ASSERT_EQ(doc["inputs"].GetArray()[0].GetObject()["shape"].GetArray()[0].GetInt(), 1);
+    ASSERT_EQ(doc["inputs"].GetArray()[0].GetObject()["shape"].GetArray()[1].GetInt(), 10);
+
+    ASSERT_EQ(std::string(doc["outputs"].GetArray()[0].GetObject()["name"].GetString()), "a");
+    ASSERT_EQ(std::string(doc["outputs"].GetArray()[0].GetObject()["datatype"].GetString()), "FP32");
+    ASSERT_EQ(doc["outputs"].GetArray()[0].GetObject()["shape"].GetArray()[0].GetInt(), 1);
+    ASSERT_EQ(doc["outputs"].GetArray()[0].GetObject()["shape"].GetArray()[1].GetInt(), 10);
+}
+
+TEST_F(HttpRestApiHandlerTest, inferRequestWithMultidimensionalMatrix) {
```
Comment (atobiszei) [inline src/test/kfs_rest_test.cpp:211]: Either I am not seeing something here or this should be negative predict request?

Model input shape is [1,10] while we send [5,2]?

Comment (atobiszei) [reply src/test/kfs_rest_test.cpp:211]: O I see - this is not going through to gRPC handling & validation. Can we fix shape here in request to match data?

**Review by atobiszei — APPROVED** [review]

**Review by atobiszei — COMMENTED** [review]
.

File: src/test/kfs_rest_test.cpp [hunk 3]
```diff
@@ -87,6 +53,8 @@ class HttpRestApiHandlerTest : public ::testing::Test {
             (char*)"/ovms/src/test/dummy",
             (char*)"--log_level",
             (char*)"DEBUG",
+            (char*)"--batch_size",
+            (char*)"auto",
```
Comment (atobiszei) [inline src/test/kfs_rest_test.cpp:57]: You need to adjust line 63 passing argc.

**Review by michalkulakowski — APPROVED** [review]

**Pull request merged**